src/node_v8.cc

```diff
@@ -157,6 +157,18 @@ void CachedDataVersionTag(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result);
 }
 
+void SetHeapSnapshotNearHeapLimit(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsNumber());
+  Environment* env = Environment::GetCurrent(args);
+  if (env->options()->heap_snapshot_near_heap_limit == 0) {
+    Isolate* const isolate = args.GetIsolate();
+    int64_t limit = args[0].As<v8::Number>()->Value();
+    CHECK_GT(limit, 0);
+    env->options()->heap_snapshot_near_heap_limit = limit;
```

Comment:
I think instead of mutating `env->options()` here, it would be better to store the configured limit in a `env->heap_snapshot_near_heap_limit_` (which also needs to be initialized), and use that in the `Environment::InitializeDiagnostics()` and `Environment::NearHeapLimitCallback()`), so that `env->options()->heap_snapshot_near_heap_limit` stays as the limit specifically configured from the command line.

Reply:
like this `env->options()->heap_snapshot_near_heap_limit || env->get_heap_snapshot_near_heap_limit()` ?

Reply:
You can just propagate  `env->heap_snapshot_near_heap_limit_` from `options()->heap_snapshot_near_heap_limit` in the Environment constructor and switch to that in the places that currently uses `options()->heap_snapshot_near_heap_limit`, to indicate that this is a state that's not only configurable from the CLI now. Also the naming convention for getters here is usually just `env->heap_snapshot_near_heap_limit()` with the state itself as `env->heap_snapshot_near_heap_limit_`

Reply:
Done !